sympy/stats/crv_types.py

```diff
@@ -1441,46 +1441,78 @@ def GammaInverse(name, a, b):
     return rv(name, GammaInverseDistribution, (a, b))
 
 #-------------------------------------------------------------------------------
-# Gumbel distribution --------------------------------------------------------
+# Gumbel distribution (Maximum and Minimum)--------------------------------------------------------
 
 
-class GumbelDistribution(SingleContinuousDistribution):
+class GumbelDistributionMaximum(SingleContinuousDistribution):
     _argnames = ('beta', 'mu')
```

Comment:
Would it be better to define an additional argument, instead of having separate classes?

Reply:
Yeah, we can do that. I defined separate classes for decoupling the properties of `GumbelDistributionMaximum` & `GumbelDistributionMinimum`. For example, if `integrate` is unable handle some property, say for instance mean or variance or skewness, for `GumbelDistributionMaximum` but is able handle the same property for `GumbelDistributionMinimum` then we can solve the problem separately. If we will use the same class then any issue for one case will affect both the cases simultaneously. 
Waiting for your comments.

Reply:
What do you think @sidhantnagpal ?

Reply:
I am not certain about changing the name of the class. CC @Upabjojr

Reply:
If the distributions are the same, we should keep only one class.

Consider using if-clauses as a workaround (if integration fails, try another way).

Anyway, If integration fails on the sign-symmetry of the function, maybe it should be reported as a bug of the integrator.